### ISSUE TITLE
AvailabilityParser marks `iOSApplicationExtension, unavailable`-annotated properties as universally deprecated

### ISSUE THREAD

**Issue by davidme-stripe** [apple/swift-docc#450]
### Description

When generating our SDK docs with DocC, classes marked as unavailable in Application Extensions are marked as deprecated in the generated docs.

<img width="541" alt="CleanShot 2023-01-04 at 15 08 43@2x" src="https://user-images.githubusercontent.com/52758633/210667061-0b79c3b1-f8de-476c-9d90-8648d5fbe634.png">

### Steps To Reproduce

[Generated DocC page for PaymentSheet.FlowController](https://davidme-stripe.github.io/docc-test/stripe-ios/stripepaymentsheet/documentation/stripepaymentsheet/paymentsheet)
[Class definition for PaymentSheet.FlowController](https://github.com/stripe/stripe-ios/blob/2a04cc683c4fa02d35f4017d282e51d7538dab79/StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFlowController.swift#L41)

If we define our class:
```
@available(iOSApplicationExtension, unavailable)
@available(macCatalystApplicationExtension, unavailable)
public class MyClass: NSObject { }
```

The Swift compiler will output this in the symbol graph JSON:
```
"availability":[{"domain":"macCatalystAppExtension","isUnconditionallyUnavailable":true},{"domain":"iOSAppExtension","isUnconditionallyUnavailable":true}]
```

And Swift-DocC will mark it as deprecated in the docs data JSON:
```
{"deprecated":true,"role":"symbol","title":"MyClass","fragments":[...]}
```

I think the issue is that the AvailabilityParser marks a symbol as deprecated if every Availability entry for a symbol is marked as `isUnconditionallyUnavailable`, even if the symbol is implicitly available in other domains. [This final check](https://github.com/apple/swift-docc/blob/d94139a5e64e9ecf158214b1cded2a2880fc1b02/Sources/SwiftDocC/Model/AvailabilityParser.swift#L41) seems incorrect:
```
        // Check if the symbol is unconditionally deprecated
        return availability.availability
            .allSatisfy { $0.isUnconditionallyDeprecated || $0.isUnconditionallyUnavailable || $0.deprecatedVersion != nil }
```
DocC already checks if a "universal deprecation" exists (a deprecation where `domain == nil`) [just above that line](https://github.com/apple/swift-docc/blob/d94139a5e64e9ecf158214b1cded2a2880fc1b02/Sources/SwiftDocC/Model/AvailabilityParser.swift#L36), though I think it should also check if the domain is `*`.

### Checklist

- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/apple/swift-docc/issues).

### Expected Behavior

DocC marks it as available, or maybe marks it as "not available in Application Extensions".

### Actual behavior

DocC marks the iOSApplicationExtension-unavailable class as Deprecated with a strikethrough effect.
<img width="541" alt="CleanShot 2023-01-04 at 15 08 43@2x" src="https://user-images.githubusercontent.com/52758633/210667061-0b79c3b1-f8de-476c-9d90-8648d5fbe634.png">

### Swift-DocC Version Information

Bundled in Xcode 14.1

### Swift Compiler Version Information

```shell
swift-driver version: 1.62.15 Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
```

**Comment by Kyle-Ye** [comment]
First of all, thank you for submitting such a detailed bug report.

There is a few thing I'd like to confirm:
1. Did you make "Expected Behavior" and "Actual behavior" swapped by mistake?
2. In "Swift-DocC Version Information", did you mean bundled in "Xcode 13.1"?

**Comment by Kyle-Ye** [comment]
A strange case I observe is that if you specify your class like SwiftUI's way, swift-doc will not mark it as deprecated 🤔.
```swift
@available(macOS 10.15, *)
@available(iOS, unavailable)
@available(tvOS, unavailable)
@available(watchOS, unavailable)
public class MyClass: NSObject { }
```

**Comment by Kyle-Ye** [comment]
Before we fix the issue, in your specific issue, you can workaround by using the following code snippet:
```swift
@available(iOSApplicationExtension, unavailable)
@available(iOS 13.0, *) // Or whatever version you like
public class MyClass: NSObject { }
```

**Comment by davidme-stripe** [comment]
> First of all, thank you for submitting such a detailed bug report.
> 
> There is a few thing I'd like to confirm:
> 
> 1. Did you make "Expected Behavior" and "Actual behavior" swapped by mistake?

Ah, yep! 🤦 Let me fix that.

> 2. In "Swift-DocC Version Information", did you mean bundled in "Xcode 13.1"?

Sorry, typo, I meant 14.1 (Build 14B47b).

> Before we fix the issue, in your specific issue, you can workaround by using the following code snippet:

Thanks, we'll try that! And thanks for the quick fix!

**Comment by Kyle-Ye** [comment]
Fixed by #712
